### PR TITLE
Maven repository is polluted with not existing dependencies

### LINKED ISSUE #213: Maven repository is polluted with not existing dependencies
We got this report on the Spring Tools distribution for Eclipse, but I think this is caused my the XML language server and its support for pom files:

https://github.com/spring-projects/sts4/issues/618

### PR DESCRIPTION
**Pull request by angelozerr**
Maven repository is polluted with not existing dependencies

Fixes #213

Take a sample with the user local repository `${userName}/.m2/repository`

This PR fixes #213 by:

 * setting the local repository of maven request to a dir different from the local user repository `${userName}/.m2/.lemminx-maven`. This local repository will contains all metadata files downaloded by aether when artifacts dependencies occurs.
 * register the user local repository `${userName}/.m2/repository` as a second local repository (with the property maven.repo.local.tail) to use it to find artifacts. This folder will never polluated with the aether metadata files when user is typing in the editor.

### CODE REVIEW

File: lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/searcher/LocalRepositorySearcher.java [hunk 1]
```diff
@@ -77,36 +72,54 @@ public Set<String> searchPluginGroupIds() throws IOException {
...
+
+	private static <T> CompletableFuture<Collection<T>> allOf(Collection<CompletableFuture<Collection<T>>> futures) {
+		return CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()])) //
+				.thenApply(__ -> futures.stream() //
+						.map(CompletableFuture::join) //
+						.flatMap(list -> list.stream()) //
+						.toList());
+	}
+
+	private void createCreateLocalArtifactsFuture(File localRepository) {
```
Comment (vrubezhny) [inline lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/searcher/LocalRepositorySearcher.java:107]: Can this be jsst `createLocalArtifactsFuture(...)`?

**Review by vrubezhny — COMMENTED** [review]
Some minor change

File: lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/searcher/LocalRepositorySearcher.java (hunk 1 above)
Comment (angelozerr) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/searcher/LocalRepositorySearcher.java]: fixed

**Comment by Madjosz** [comment]
Please adjust your formatter to only format actually changed lines. The commits contain much clutter where nothing relevant has changed but only some whitespace or linefeeds.

**Pull request merged**